src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -651,6 +652,9 @@ ReactDOMComponent.Mixin = {
             markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
           }
         } else {
+          if (__DEV__) {
+            ReactDOMInstrumentation.debugTool.onCreateMarkupForProperty(propKey, propValue, this._currentElement);
```

Comment:
This line is the main one that concerns me.  It seems like this call should be inside `createMarkupForProperty` such that it gets invoked whenever anyone calls it will get the `onCreateMarkupForProperty` event.  However, we should not be passing the element to `createMarkupForProperty`.

Thus, why this PR is difficult, thus my comments in https://github.com/facebook/react/issues/6062#issuecomment-185528241 and https://github.com/facebook/react/issues/6062#issuecomment-185535439